components/dialog/dialog.js

```diff
@@ -90,6 +97,19 @@ class Dialog extends LocalizeStaticMixin(DialogMixin(LitElement)) {
 	}
 
 	render() {
+
+		let content;
+		if (this.async && this.asyncState !== asyncStates.complete) {
+			content = html`
+				<div class="d2l-dialog-content-loading">
+					<d2l-loading-spinner size="100"></d2l-loading-spinner>
+				</div>
+				<div style="display: none;"><slot></slot></div>
```

Comment:
The slotted async content is still rendered into the DOM so that we can listen for it's async events, however we set `display: none` so that it does not add any extra height/whitespace below the spinner.

Reply:
Nit: seems like there could be a slight optimization here somehow so that the `<div><slot></slot></div>` doesn't need to be repeated in both branches of the `if`-statement.

Maybe using a CSS class and `classMap` to hide the `<div>`?

```
let asyncContent;
let contentClasses = {};
if (this.async && this.asyncState !== asyncStates.complete) {
    contentClasses = {
			'd2l-dialog-content-hidden': true
		};
    asyncContent = html`<div class="d2l-dialog-content-loading><d2l-loading-spinner>...`;
}
```

```

Reply:
Yeah I can pull that out.  I think each `${some-other-template}` might might result in start/end comment nodes generated by Lit (it uses them) so I was sorta just avoiding those.

Reply:
I've refactored this to remove the duplication.  These are the comments I was referring to... 🤷‍♂ ...
![image](https://user-images.githubusercontent.com/9042472/72166681-b2bc8900-3397-11ea-97af-049db50be793.png)